src/lib/utils/index.ts

```diff
@@ -96,12 +96,22 @@ export function partitionCompareWith<T>(
   return result
 }
 
-export const isToken = (assetReference: AssetReference | string) => {
-  return !Object.values(ASSET_REFERENCE).includes(assetReference as AssetReference)
+export const isToken = (assetId: AssetId) => {
```

Comment:
oh man tyvm 🐐 got burned by this so many times (i.e `string` union on the left, and since `AssetId` is a string,  so wouldn't be caught by tsc